internal/venice-common/src/main/java/com/linkedin/venice/pubsub/PubSubPositionDeserializer.java

```diff
@@ -59,31 +59,50 @@ public PubSubPosition convertToPosition(PubSubPositionWireFormat positionWireFor
     return factory.createFromWireFormat(positionWireFormat);
   }
 
-  public PubSubPosition convertToPosition(byte[] positionWireFormatBytes) {
+  public PubSubPosition toPosition(byte[] positionWireFormatBytes) {
     if (positionWireFormatBytes == null) {
       throw new IllegalArgumentException("Cannot deserialize null wire format position");
     }
-    InternalAvroSpecificSerializer<PubSubPositionWireFormat> wireFormatSerializer =
-        AvroProtocolDefinition.PUBSUB_POSITION_WIRE_FORMAT.getSerializer();
-    PubSubPositionWireFormat wireFormat = wireFormatSerializer.deserialize(positionWireFormatBytes, null);
-    return convertToPosition(wireFormat);
+    PubSubPositionWireFormat wireFormat =
+        PUBSUB_POSITION_WIRE_FORMAT_SERIALIZER.deserialize(positionWireFormatBytes, null);
+    return toPosition(wireFormat);
   }
 
-  public PubSubPosition convertToPosition(ByteBuffer positionWireFormatBytes) {
+  public PubSubPosition toPosition(ByteBuffer positionWireFormatBytes) {
     if (positionWireFormatBytes == null) {
       throw new IllegalArgumentException("Cannot deserialize null wire format position");
     }
-    InternalAvroSpecificSerializer<PubSubPositionWireFormat> wireFormatSerializer =
-        AvroProtocolDefinition.PUBSUB_POSITION_WIRE_FORMAT.getSerializer();
-    PubSubPositionWireFormat wireFormat = wireFormatSerializer.deserialize(positionWireFormatBytes.array(), null);
-    return convertToPosition(wireFormat);
+    PubSubPositionWireFormat wireFormat =
+        PUBSUB_POSITION_WIRE_FORMAT_SERIALIZER.deserialize(positionWireFormatBytes.array(), null);
```

Comment:
Should we extract byte array from it incase it is not starting from 0?

Reply:
Good point. Updated. Thanks!